volunteer_base/models/volunteer_engagement.py

```diff
@@ -0,0 +1,191 @@
+        help="In this field the end date of an engagement can be saved."
+    )
+    state_onhold_until = fields.Date(
+        string='On hold until',
+        tracking=True,
+        track_visibility='always',
+        help="In this field that date when a volunteering engagmeent was set on hold"
+    )
+    date_cancel = fields.Date(
+        string='Cancel date',
+        help="Date when Volunteer requested to cancel the engagement"
+    )
+    cancel_reason_id = fields.Many2one(
+        string='Cancel reason',
+        comodel_name='engagement.cancel.reason',
+        index=True,
+        track_visibility='onchange',
+        help="The reason why a volunteer caneceled an engagement"
+    )
+    onhold_reason_id = fields.Many2one(
+        string='On hold reason',
+        comodel_name='engagement.onhold.reason',
+        index=True,
+        track_visibility='onchange',
+        help="The reason why a volunteer puts an engagement on hold"
+    )
+
+    filerable_groupable_fields = ['state','state_onhold_until','start_date','end_date']
+
+    _sql_constraints = [
+        ('volunteer_engagement_unique_name', 'UNIQUE (name)',
+         'The Engagement Reference must be unique!'),
+    ]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get('name', '/') == '/':
+                vals['name'] = self.env['ir.sequence'].next_by_code('volunteer.engagement')
+        return super(VolunteerEngagement, self).create(vals_list)
+
+    @api.multi
+    def copy(self, default=None):
+        if default is None:
+            default = {}
+        default['name'] = self.env['ir.sequence'].next_by_code('volunteer.engagement')
+        return super(VolunteerEngagement, self).copy(default)
+
+    @api.multi
+    def action_volunteer_engagement_active(self):
+        if self.state == 'new':
+            project = self.volunteer_project_id
+            if project.timeframe == 'ongoing':
+                date_start = fields.Date.today()
+                date_end = fields.Date.today() + self.get_relative_delta(
+                project.engagement_default_duration_type, project.engagement_default_duration)
+            elif project.timeframe == 'period':
+                date_start = datetime.combine(project.date_start, time.min)
+                date_end = datetime.combine(project.date_end, time.min)
+
+            self.write({
+                'state': 'active',
+                'start_date': date_start,
+                'end_date': date_end})
+
+    @api.multi
+    def action_volunteer_engagement_success(self):
+        self.write({
+            'state': 'success',
+            'end_date': fields.Date.today()
+        })
+        
+    @api.multi
+    def action_volunteer_engagement_reset(self):
+        return self.write({'state': 'new'})
+
+    @api.onchange('start_date', 'end_date')
+    def _onchange_start_end_date(self):
+        if self.start_date != False and self.end_date != False:
```

Comment:
Code style: use "is" or "is not" for boolean comparisons

```suggestion
        if self.start_date is not False and self.end_date is not False:
```